src/specific/display.c

```diff
@@ -10,10 +10,66 @@
 
 #include <stdlib.h>
 
+// The screen resolution is controlled by two variables are indices
+// within an array of predefined screen resolutions.
+// Actual screen resolution, sometimes different from the game
+// resolution (during FMVs, main menu sequence, static pictures etc.)
+static int32_t HiRes = 0;
+// The resolution to render the game in. This is what gets saved in
+// settings and the like.
+static int32_t GameHiRes = 0;
+
+bool SetGameScreenSizeIdx(int32_t idx)
+{
+    if (idx >= 0 && idx < RESOLUTIONS_SIZE) {
+        GameHiRes = idx;
+        return true;
+    }
+    return false;
+}
+
+bool SetPrevGameScreenSize()
+{
+    if (GameHiRes - 1 >= 0) {
+        GameHiRes--;
+        return true;
+    }
+    return false;
+}
+
+bool SetNextGameScreenSize()
+{
+    if (GameHiRes + 1 < RESOLUTIONS_SIZE) {
```

Comment:
RESOLUTIONS_SIZE is a constant, so do GameHiRes < RESOLUTIONS_SIZE - 1 as the -1 can be done at compile time while the + 1 has to be done at run time.

Reply:
Microoptimization that saves maybe a few nanoseconds at the cost of readability